p4-16/spec/P4-16-spec.mdk

```diff
@@ -1857,10 +1860,21 @@ Boolean values are not integers or bit-strings.
 
 ### Strings { #sec-strings }
 
-P4 offers no support for string processing. The only strings that can
-appear in a P4 program are constant string literals, described in
-Section [#sec-string-literals]. String literals can only be used in
-annotations (described in Section [#sec-annotations]). For example,
+The type `string` represents strings.  There are no operations on
+string values; one cannot declare variables with a `string` type.
+Parameters with type `string` can be only directionless (see Section
+[#sec-calling-convention]).  P4 does not support string manipulation
+in the dataplane; the `string` type is only allowed for denoting
+compile-time constant string values.  These may be useful, for
+example, a specific target architecture may support an extern function
+for logging with the following signature:
```

Comment:
Reference the string encoding from the 'String literals' section or tighten it more to UTF8?

Reply:
I don't think it's necessary: we don't say anything else about string literals in this section.